service-base/src/main/java/org/eclipse/hono/service/auth/X509CertificateChainValidator.java

```diff
@@ -28,15 +27,16 @@
 public interface X509CertificateChainValidator {
 
     /**
-     * Validates a certificate path based on a given trust anchor.
+     * Validates a certificate path based on a list of trust anchors.
      * 
      * @param chain The certificate chain to validate. The end certificate
      *              must be at position 0.
-     * @param trustAnchor The trust anchor to use for validating the chain.
-     * @return A completed future if the path is valid (according to the implemented tests).
-     *         Otherwise, the future will be failed with a {@link CertificateException}.
-     * @throws NullPointerException if any of the parameters are {@code null}.
-     * @throws IllegalArgumentException if the chain is empty.
+     * @param trustAnchors The list of trust anchors to use for validating the chain.
+     * @return A composite future, succeeded when the certificate path is valid for any the trust anchors.
+     *         Otherwise, the composite future is failed when all the trust anchors cannot be validated by the certificate path.
+     *
+     * @throws NullPointerException if any of the paramerters are {@code null}.
+     * @throws IllegalArgumentException if the chain and/or trust anchors is empty.
      */
-    Future<Void> validate(List<X509Certificate> chain, TrustAnchor trustAnchor);
+    CompositeFuture validate(List<X509Certificate> chain, List<TrustAnchor> trustAnchors);
```

Comment:
Well, I was actually thinking about simply adding another variant of this method that accepts a List of anchors ...

Reply:
since one can create a list containing a single trust anchor, don't you think that this is just a general case of the current call: e.g `validate(chain, List.of(trustAnchor))`?

Reply:
That is true but keeping the existing method does not break any client code and also is a little more convenient. I would also rather have a `Future<Void>` as the return value instead of a `CompositeFuture`.

Reply:
@sophokles73 done